cmd/start.go

```diff
@@ -93,31 +99,123 @@ func runContainer(cmd *cobra.Command, args []string) {
 		"CEPH_DEMO_UID=" + CephNanoUID,
 		"NETWORK_AUTO_DETECT=4",
 		"CEPH_DAEMON=demo",
-		"DEMO_DAEMONS=mon,mgr,osd,rgw"}
+		"DEMO_DAEMONS=mon,mgr,osd,rgw",
+	}
+
+	volumeBindings := []string{
+		WorkingDirectory + ":" + TempPath,
+	}
 
 	ressources := container.Resources{
 		Memory:   536870912, // 512MB
 		NanoCPUs: 1,
 	}
 
+	volumes := map[string]struct{}{
+		"/etc/ceph":     struct{}{},
+		"/var/lib/ceph": struct{}{},
+	}
+
+	if len(Data) != 0 {
+		testDev, err := GetFileType(Data)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if testDev != "blockdev" && testDev != "directory" {
+			log.Fatalf("We only accept a directory or a block device, however the specified file type is a %s", testDev)
+		}
+		if testDev == "directory" {
+			testEmptyDir := IsEmpty(Data)
+			if !testEmptyDir {
+				log.Fatal(Data + " is not empty, doing nothing.")
+			}
+			envs = append(envs, "OSD_PATH="+Data)
+			volumeBindings = append(volumeBindings, Data+":"+Data)
+		}
+		if testDev == "blockdev" {
+			meUserName, meID := whoAmI()
+			if meID != "0" {
+				log.Fatal("Hey " + meUserName + "! Run me as 'root' when using a block device.")
+			}
+			// We don't have the logic to do the introspection without using blkid.
+			// Unfortunately, blkid is not available on macOS or Windows
+			if runtime.GOOS == "windows" || runtime.GOOS == "darwin" {
+				log.Fatal("Operating system: " + runtime.GOOS + " is not supported in the scenario")
+			}
+			// We run a couple of test here to ensure the device can be used:
+			// 1. test of the device is accessed by a process (open it with O_EXCL)
+			// 2. test if the device has a partition table and/or a filesystem
+			// 3. test if they are partitions in that partition table (you can have a partition table with 0 partitions)
+
+			// First test: is the device opened by a process?!«
+			testDevOpen, _ := DeviceOpened(Data)
+			if testDevOpen {
+				log.Fatal(Data + " is accessed by another process, doing nothing.")
+			}
+
+			// Second test: search for filesystem and partition table
+			diskFormat := GetDiskFormat(Data)
+			lines := strings.Split(diskFormat, "\n")
+			var fstype, pttype string
+			for _, l := range lines {
+				if len(l) <= 0 {
+					// Ignore empty line.
+					continue
+				}
+				cs := strings.Split(l, "=")
+				if len(cs) != 2 {
+					log.Fatal("blkid returns invalid output: " + diskFormat + ". This potentially mean no partition label on your disk.")
+				}
+				// TYPE is filesystem type, and PTTYPE is partition table type, according
+				// to https://www.kernel.org/pub/linux/utils/util-linux/v2.21/libblkid-docs/.
+				if cs[0] == "TYPE" {
+					fstype = cs[1]
+					log.Fatal(Data + " has a filesystem: " + fstype + ", doing nothing.")
+				} else if cs[0] == "PTTYPE" {
```

Comment:
If the given disk is an msdos extended partition will return PTTYPE and the following parted will fail on it.

erwan@R1:~/Devel/ceph/src (evelu-readonly)$ sudo blkid /dev/sda4
/dev/sda4: PTTYPE="dos" PARTUUID="ee3a7940-04"

erwan@R1:~/Devel/ceph/src (evelu-readonly)$ sudo parted -sm /dev/sda4 print
Erreur: La partition ne peut pas être en dehors du disque !
BYT;
/dev/sda4:1024B:unknown:512:512:unknown:Inconnu:;

Have you made a try on a GPT partition table ?

Reply:
GPT report PTYPE GPT